lib/components/MarkdownInterpreter.tsx

```diff
@@ -1,151 +1,257 @@
-import type { ReactNode } from 'react'
 import { tw } from '../twind'
 
-export type MarkdownInterpreterProps = {
-  text: string,
-  commandStart?: string,
-  open?: string,
-  close?: string
-}
+const astNodeModifierTypes = ['none', 'italic', 'bold', 'underline', 'font-space', 'primary', 'warn', 'positive', 'negative'] as const
+type ASTNodeModifierType = typeof astNodeModifierTypes[number]
 
-type Marker = {
-  id: string,
-  className: string
+const astNodeInserterType = ['helpwave'] as const
+type ASTNodeInserterType = typeof astNodeInserterType[number]
+type ASTNodeDefaultType = 'text'
+
+type ASTNode = {
+  type: ASTNodeModifierType,
+  children: ASTNode[]
+} | {
+  type: ASTNodeInserterType
+} | {
+  type: ASTNodeDefaultType,
+  text: string
 }
 
-type Replace = {
-  id: string,
-  replacement: string | ReactNode
+export type ASTNodeInterpreterProps = {
+  node: ASTNode,
+  isRoot?: boolean
 }
+export const ASTNodeInterpreter = ({
+  node,
+  isRoot = false
+}: ASTNodeInterpreterProps) => {
+  switch (node.type) {
+    case 'text':
+      return isRoot ? <span>{node.text}</span> : node.text
+    case 'helpwave':
+      return (<span className={tw('font-bold font-space no-underline')}>helpwave</span>)
+    case 'none':
+      return isRoot ? <span>{node.children.map((value, index) => <ASTNodeInterpreter key={index} node={value}/>)}</span> :
+        <>{node.children.map((value, index) => <ASTNodeInterpreter key={index} node={value}/>)}</>
+    case 'bold':
+      return <b>{node.children.map((value, index) => <ASTNodeInterpreter key={index} node={value}/>)}</b>
+    case 'italic':
+      return <i>{node.children.map((value, index) => <ASTNodeInterpreter key={index} node={value}/>)}</i>
+    case 'underline':
+      return (
+        <span className={tw('underline')}>{node.children.map((value, index) => (
+          <ASTNodeInterpreter key={index}
+                              node={value}/>
+        ))}</span>
```

Comment:
A particular reason why we are not using `<u/>` here?

Reply:
no, I will change it